llvm/include/llvm/Support/GenericDomTree.h

```diff
@@ -885,6 +874,12 @@ class DominatorTreeBase {
         NewBBIDom = findNearestCommonDominator(NewBBIDom, PredBlocks[i]);
     }
 
+    bool NewBBDominatesNewBBSucc =
```

Comment:
is there a reason this code moved from 20 lines above to down here? Might be worth doing in a separate commit if that's intentional, because it's less obviously NFC than the simple transformation to use `llvm::none_of`?

Reply:
Evaluating `NewBBDominatesNewBBSucc` looks independent to finding `NewBBIDom`. Since we don't need `NewBBDominatesNewBBSucc` on the path that `if (!NewBBIDom) return;` is triggered, it was moved down. (Reverted now.) I can do it in a separate PR if that sounds reasonable.

Reply:
Yeah, just makes things a bit clearer/easier to review, thanks!